GSoC_19/posts/websocket-passive-scan-using-scripts-with-zap.org

```diff
@@ -0,0 +1,851 @@
+
+   ECMAScript / Javascript (using Nashorn engine, included by default)
+
+   Zest https://developer.mozilla.org/en-US/docs/zest (included by default)
+
+   Groovy http://groovy-lang.org/
+
+   Python http://www.jython.org
+
+   Ruby - http://jruby.org/
+
+   and many more...
+   
+   /from [[https://github.com/zaproxy/zap-core-help/wiki/HelpAddonsScriptsScripts][wiki]]/
+   #+END_QUOTE
+
+   Referring to scripts console add-on we are referring to the ability that the
+   user has to write scripts which implement various behaviors like scans, fuzzing etc.
+   Scripts Console add-on has a panel where the scripts are displayed. The panel
+   has a tree structure, which has two branches, the =Scripts= branch and
+   =Templates=. In both branches there are some folders which are referring on
+   different kind of scripts. As you can see those folders in both branches have
+   the same names.  
+
+   #+ATTR_HTML: :align center :width 300 height 800
+   #+CAPTION: Scripts Tree Sidebar
+   [[../../images/pscan-tutorial/scripts_tree.png]]
+
+   ZAP defines =Templates= in order to be used as base scripts. As we will see in
+   a while every kind of script should implement a specific functionality, named
+   rule. In addition, =Templates= are not runnable, enablable or directly
+   editable, they are used only as templates. In order to use a template, right
+   click on it and then hit =New script...=. 
+
+   **Hands On Section:**
+   Go to the template script =Base64 Disclosure.js=, right click on it, change
+   =Script Name= to =MyBase64 Disclosure.js= and hit save. See what happened.
+
+   =Scripts=, on the other hand, are the actual scripts which are run by ZAP.
+   There are a good many possibilities in which we can use scripts,
+   however, in this guide we are interested in WebSocket Passive Scanning.
+   Passive Scripts have two states, enable and disable. When a script is disabled means
+   that it is not running at all. Other options like =duplicate= script or =remove= script are
+   available by right clicking on them. **You should always remember that Right**
+   **Click in ZAP is your best friend.** By default, the folders under =Scripts=
+   are empty, except from the =WebSocket Passive Rules=. If you want to explore
+   more the scripting techniques and add some rules in ZAP, I totally recommend
+   to install =Community Scripts= from =marketplace=. 
+   
+   **Hands On Section:**
+   Go to the script that was created before. If it is not disabled, then disable it
+   using right click. Navigate over the code and find the following line: 
+   
+   #+BEGIN_SRC js
+   PRINT_RESULTS = false;
+   #+END_SRC
+   
+   make it =true=, enable and save the script. 
+
+   Passive scan scripts are running against all messages, as result we do not have to
+   run them explicitly. So the only we have to do is to enable them, and when
```

Comment:
`as result we do` &gt; `as a result we do`

`So the only we have to do is` &gt; `So the only thing we have to do is`